opensearch_py_ml/ml_commons/model_profile.py

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+from typing import Optional
+
+from opensearchpy import OpenSearch
+
+from opensearch_py_ml.ml_commons.ml_common_utils import ML_BASE_URI
+
+
+class ModelProfile:
```

Comment:
I don't think we need a separate class for this api. This is a super simple api. And also I think we can consolidate all these 3 methods into 1 method. Why can't we work on [this PR](https://github.com/opensearch-project/opensearch-py-ml/pull/119/files)?

Reply:
Sure. I'll scrap the PR and reuse the code. Only reason why I didn't reuse it is that, I felt, from a user perspective, it would be better to give three methods with names that explain what results they will get. But, i understand we can just use one as well.